tests/test_presentation.py

```diff
@@ -0,0 +1,199 @@
+        xml_linking_object_type = xml_obj.linkingObjectIdentifier.linkingObjectIdentifierType.text
+
+        # Get the newly created LinkObject from the database.
+        linking_object = event.linking_objects.get(object_identifier=xml_linking_object_id)
+
+        # Assert that the database version matches the values defined in the XML.
+        assert xml_linking_object_id == linking_object.object_identifier
+        assert xml_linking_object_type == linking_object.object_type
+
+    @pytest.mark.xfail(reason='Try block always catches a NameError before able to return.')
+    def test_existing_event_identifier_results_in_HttpResponse(self, event_xml):
+        identifier, xml = event_xml
+        tree = etree.fromstring(xml)
+
+        factories.EventFactory.create(event_identifier=identifier)
+        event_or_response = presentation.premisEventXMLToObject(tree)
+
+        assert isinstance(event_or_response, HttpResponse)
+
+    def test_new_identifier_created_if_not_valid_uuid4(self, event_xml):
+        _, xml = event_xml
+
+        # Replace the existing hex identifier with a non-hex identifier.
+        invalid_identifier = 'invalid-hex'
+        xml_obj = objectify.fromstring(xml)
+        xml_obj.eventIdentifier.eventIdentifierValue = invalid_identifier
+
+        tree = objectify_to_etree(xml_obj)
+
+        # Create an Event with the same non-hex identifier.
+        factories.EventFactory.create(event_identifier=invalid_identifier)
+        event = presentation.premisEventXMLToObject(tree)
+
+        # Make sure the returned Event is given a new hex identifier.
+        assert event.event_identifier != invalid_identifier
+        assert uuid.UUID(event.event_identifier, version=4)
+
+    @pytest.mark.xfail(reason='Validation error is raised on save(). The exception '
+                              'this function tests will never be raised.')
+    def test_invalid_datetime_string_raises_exception(self, event_xml):
+        _, xml = event_xml
+
+        # Replace the datetime string with an invalid datetime string.
+        xml_obj = objectify.fromstring(xml)
+        xml_obj.eventDateTime = 'invalid-datetime'
+
+        tree = objectify_to_etree(xml_obj)
+
+        with pytest.raises(Exception) as e:
+            presentation.premisEventXMLToObject(tree)
+
+        # Since the base Exception is raised from the function, we want to make
+        # sure it is the correct Exception, so now check the error message and
+        # see if it matches the expected message.
+        assert 'Unable to parse' in str(e)
+        assert 'into datetime object' in str(e)
+
+    def test_datetime_that_has_decimal_seconds(self, event_xml):
+        """Test that a datetime string that contains a decimal second is properly
+        converted to a datetime object.
+        """
+        _, xml = event_xml
+
+        # Replace the datetime string with a valid datetimestring that
```

Comment:
Should "datetimestring be "datetime" string"? Other than that, I give the PR :+1: .